src/main/java/de/hysky/skyblocker/skyblock/item/tooltip/adders/BitsHelper.java

```diff
@@ -0,0 +1,448 @@
+package de.hysky.skyblocker.skyblock.item.tooltip.adders;
+
+import de.hysky.skyblocker.config.SkyblockerConfigManager;
+import de.hysky.skyblocker.skyblock.item.tooltip.TooltipInfoType;
+import de.hysky.skyblocker.skyblock.itemlist.ItemRepository;
+import de.hysky.skyblocker.utils.ItemUtils;
+import de.hysky.skyblocker.utils.container.SimpleContainerSolver;
+import de.hysky.skyblocker.utils.container.TooltipAdder;
+import de.hysky.skyblocker.utils.render.gui.ColorHighlight;
+import it.unimi.dsi.fastutil.ints.Int2ObjectMap;
+import it.unimi.dsi.fastutil.ints.Int2ObjectOpenHashMap;
+import net.minecraft.client.MinecraftClient;
+import net.minecraft.client.gui.screen.ingame.HandledScreen;
+import net.minecraft.item.ItemStack;
+import net.minecraft.screen.ScreenHandler;
+import net.minecraft.screen.slot.Slot;
+import net.minecraft.text.Text;
+import net.minecraft.util.Formatting;
+import net.minecraft.util.Pair;
+import net.minecraft.util.Util;
+import org.intellij.lang.annotations.Language;
+import org.jetbrains.annotations.Nullable;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.text.NumberFormat;
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class BitsHelper extends SimpleContainerSolver implements TooltipAdder {
+    private static final Pattern BITS_PATTERN = Pattern.compile("Cost (?<amount>[\\d,]+) Bits");
+    private static final Pattern CATEGORY_PATTERN = Pattern.compile("Click to browse!");
+    @Language("RegExp") private static final String TITLE_PATTERN = ".*(?:Community Shop|Bits Shop).*";
+    private static final NumberFormat DECIMAL_FORMAT = NumberFormat.getInstance(Locale.US);
+    private static final Logger LOGGER = LoggerFactory.getLogger("Skyblocker Bits");
+    private static final String LOGS_PREFIX = "[Skyblocker Bits Helper] ";
+
+    public static final BitsHelper INSTANCE = new BitsHelper();
+
+    public BitsHelper() {
+        super(TITLE_PATTERN);
+    }
+
+    private Map<String, Pair<String, Long>> categoryOutput = new HashMap<>();
```

Comment:
should use an `ObjectLongPair<String>` to avoid boxing

Reply:
we are saving kilobyte of ram with this one 🔥🔥🔥